test/core/end2end/tests/retry_streaming.cc

```diff
@@ -59,345 +45,129 @@ static std::unique_ptr<CoreTestFixture> begin_test(
-  op->data.send_status_from_server.status_details = &status_details;
-  op++;
-  error = grpc_call_start_batch(s, ops, static_cast<size_t>(op - ops),
-                                grpc_core::CqVerifier::tag(205), nullptr);
-  GPR_ASSERT(GRPC_CALL_OK == error);
-  cqv.Expect(grpc_core::CqVerifier::tag(205), true);
-  cqv.Expect(grpc_core::CqVerifier::tag(1), true);
-  cqv.Verify();
-
-  GPR_ASSERT(status == GRPC_STATUS_ABORTED);
-  GPR_ASSERT(0 == grpc_slice_str_cmp(details, "xyz"));
-  GPR_ASSERT(0 == grpc_slice_str_cmp(call_details.method, "/service/method"));
-  GPR_ASSERT(was_cancelled == 0);
-
-  GPR_ASSERT(channelz_channel != nullptr);
+  CoreEnd2endTest::IncomingCloseOnServer client_close2;
+  s2.NewBatch(205)
+      .RecvCloseOnServer(client_close2)
+      .SendInitialMetadata({})
+      .SendMessage("quux")
+      .SendStatusFromServer(GRPC_STATUS_ABORTED, "xyz", {});
+  Expect(205, true);
+  Expect(1, true);
+  Step();
+  EXPECT_EQ(server_status.status(), GRPC_STATUS_ABORTED);
+  EXPECT_EQ(server_status.message(), "xyz");
+  EXPECT_EQ(s.method(), "/service/method");
+  EXPECT_FALSE(client_close.was_cancelled());
+  EXPECT_NE(channelz_channel, nullptr);
   std::string json = channelz_channel->RenderJsonString();
-  gpr_log(GPR_INFO, "%s", json.c_str());
-  GPR_ASSERT(json.find("\"trace\"") != json.npos);
-  GPR_ASSERT(json.find("\"description\":\"Channel created\"") != json.npos);
-  GPR_ASSERT(json.find("\"severity\":\"CT_INFO\"") != json.npos);
-  GPR_ASSERT(json.find("Resolution event") != json.npos);
-  GPR_ASSERT(json.find("Created new LB policy") != json.npos);
-  GPR_ASSERT(json.find("Service config changed") != json.npos);
-  GPR_ASSERT(json.find("Address list became non-empty") != json.npos);
-  GPR_ASSERT(json.find("Channel state change to CONNECTING") != json.npos);
-
-  grpc_slice_unref(details);
-  grpc_metadata_array_destroy(&initial_metadata_recv);
-  grpc_metadata_array_destroy(&trailing_metadata_recv);
-  grpc_metadata_array_destroy(&request_metadata_recv);
-  grpc_call_details_destroy(&call_details);
-  grpc_byte_buffer_destroy(request_payload);
-  grpc_byte_buffer_destroy(request2_payload);
-  grpc_byte_buffer_destroy(request3_payload);
-  grpc_byte_buffer_destroy(response_payload);
-  GPR_ASSERT(byte_buffer_eq_slice(request_payload_recv, request_payload_slice));
-  grpc_byte_buffer_destroy(request_payload_recv);
-  GPR_ASSERT(
-      byte_buffer_eq_slice(request2_payload_recv, request2_payload_slice));
-  grpc_byte_buffer_destroy(request2_payload_recv);
-  GPR_ASSERT(
-      byte_buffer_eq_slice(request3_payload_recv, request3_payload_slice));
-  grpc_byte_buffer_destroy(request3_payload_recv);
-  grpc_byte_buffer_destroy(response_payload_recv);
-
-  grpc_call_unref(c);
-  grpc_call_unref(s);
-}
-
-void retry_streaming(const CoreTestConfiguration& config) {
-  GPR_ASSERT(config.feature_mask & FEATURE_MASK_SUPPORTS_CLIENT_CHANNEL);
-
-  test_retry_streaming(config);
+  EXPECT_THAT(json, HasSubstr("\"trace\""));
```

Comment:
Now that this has been converted to gmock, I bet we could replace these individual substring checks with a regex check that would be more readable.  Probably best not to change that in this PR, but please add a TODO(roth) about this.

Reply:
done